### PR TITLE
Added section about node autoscaler to cluster-management design doc.

### PR DESCRIPTION
**Pull request by jszczepkowski**
Added section about node autoscaler to node design doc.

### CODE REVIEW

**Comment by jszczepkowski** [comment]
CC @piosz

**Comment by jszczepkowski** [comment]
Part of #11748.

**Comment by jszczepkowski** [comment]
CC @fgrzadkowski

**Comment by fgrzadkowski** [comment]
Shouldn't this go somewhere in cluster-management.md?

File: docs/admin/node.md [hunk 1]
```diff
@@ -258,6 +258,29 @@ Set the `cpu` and `memory` values to the amount of resources you want to reserve
 Place the file in the manifest directory (`--config=DIR` flag of kubelet).  Do this
 on each kubelet where you want to reserve resources.
 
+### Horizontal auto-scaling of nodes
+
+If you are using GCE, you can configure your cluster so that the number of nodes will be automatically scaled based on their utilization.
```
Comment (davidopp) [inline docs/admin/node.md]: s/utilization/CPU utilization/

File: docs/admin/node.md [hunk 2]
```diff
@@ -258,6 +258,29 @@ Set the `cpu` and `memory` values to the amount of resources you want to reserve
...
+Before setting-up cluster by ```kube-up.sh```, you can set ```KUBE_ENABLE_NODE_AUTOSCALER``` environment variable to ```true``` and export it.
```
Comment (davidopp) [inline docs/admin/node.md]: s/setting up/setting up the/

File: docs/admin/node.md [hunk 3]
```diff
@@ -258,6 +258,29 @@ Set the `cpu` and `memory` values to the amount of resources you want to reserve
...
+The script will create an autoscaler for the instance group managing your nodes.
+
+The autoscaler will try to  maintain the average utilization of nodes within the cluster close to the target value.
```
Comment (davidopp) [inline docs/admin/node.md]: s/utilization/CPU utilization/

File: docs/admin/node.md [hunk 4]
```diff
@@ -258,6 +258,29 @@ Set the `cpu` and `memory` values to the amount of resources you want to reserve
...
+The target value can be configured by ```KUBE_TARGET_NODE_UTILIZATION``` environment variable (default: 0.7).
```
Comment (davidopp) [inline docs/admin/node.md]: I assume you can't change this after running kube-up.sh -- you should say that.

File: docs/admin/node.md (hunk 1 above)
Comment (davidopp) [inline docs/admin/node.md]: sorry, didn't realize it was also memory -- so s/utilization/CPU and memory utilization/

File: docs/admin/node.md (hunk 3 above)
Comment (davidopp) [inline docs/admin/node.md]: sorry, didn't realize it was also memory -- so s/utilization/CPU and memory utilization/

File: docs/admin/node.md [hunk 5]
```diff
@@ -258,6 +258,29 @@ Set the `cpu` and `memory` values to the amount of resources you want to reserve
...
+The node utilization is the total node's cpu/memory usage (OS + k8s + user load) divided by the node's capacity.
```
Comment (davidopp) [inline docs/admin/node.md]: Maybe say a little more about how you turn this into a single number -- e.g. is it
(CPU load/CPU capacity + memory load/memory capacity)/2 ?

**Comment by davidopp** [comment]
I agree with @fgrzadkowski this should go in cluste-rmanagement.md

File: docs/admin/node.md (hunk 1 above)
Comment (jszczepkowski) [inline docs/admin/node.md]: Done.

File: docs/admin/node.md (hunk 2 above)
Comment (jszczepkowski) [inline docs/admin/node.md]: Done.

File: docs/admin/node.md (hunk 3 above)
Comment (jszczepkowski) [inline docs/admin/node.md]: Done.

File: docs/admin/node.md (hunk 4 above)
Comment (jszczepkowski) [inline docs/admin/node.md]: Done.

File: docs/admin/node.md (hunk 5 above)
Comment (jszczepkowski) [inline docs/admin/node.md]: Done.

**Comment by jszczepkowski** [comment]
> I agree with @fgrzadkowski this should go in cluste-rmanagement.md

I thought node.md is better, as in node.md is written how to resize the cluster:
https://github.com/kubernetes/kubernetes/blob/master/docs/admin/node.md#self-registration-of-nodes

Still, I you wish I can move my sections to cluster-management.md. In such case, it is probably also reasonable to move there the info about resize from node.md (I can do it too).

**Comment by jszczepkowski** [comment]
@davidopp Comments applied. PTAL

**Comment by davidopp** [comment]
Yeah, I still think this should go in cluster-management.md. It will be more discoverable there. Moving the resize info along with it is fine. Probably you should add pointers between these two files so people are more likely to read both. But people are more likely to read cluster-management.md, so we should put the important operational-oriented stuff there, like autoscaling and resizing.

**Comment by jszczepkowski** [comment]
@davidopp 
I've moved sections about manual&automatic resize to cluster-management.md. PTAL

**Comment by davidopp** [comment]
LGTM

Maybe in a followup PR you can add pointers between cluster-management.md and node.md so people who find one of the docs will also read the other.

**Pull request merged**